### PR TITLE
[Dec/Sub] Add octet-stream decoder subplugin. @open sesame 12/10 14:24

### PR DESCRIPTION
**Pull request by gichan-jang**
Decode tensors to octet stream.
If tensor format is flexible, remove the header.

**Self evaluation:**
1. Build test: [*]Passed [ ]Failed [ ]Skipped
2. Run test: [*]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: ext/nnstreamer/tensor_decoder/tensordec-octetstream.c [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    GstMemory *mem = NULL;
+
+    if (is_flexible) {
+      gst_tensor_meta_info_parse_header (&meta, input[i].data);
+      offset = gst_tensor_meta_info_get_header_size (&meta);
+      data_size = gst_tensor_meta_info_get_data_size (&meta);
+    } else {
+      data_size = gst_tensors_info_get_size (&config->info, i);
+    }
+    mem_data = g_memdup (input[i].data, input[i].size);
```
Comment (jaeyun-jung) [inline ext/nnstreamer/tensor_decoder/tensordec-octetstream.c:98]: 1. allocate memory with data only, not all input
2. use macro _g_memdup in nnstreamer_util.h
```
mem_data = _g_memdup ((guint8) input[i].data + offset, data_size);
```

File: jni/Android.mk [hunk 2]
```diff
@@ -218,4 +218,20 @@ LOCAL_C_INCLUDES += $(GST_HEADERS_COMMON)
 
 include $(BUILD_SHARED_LIBRARY)
 
+include $(CLEAR_VARS)
+LOCAL_MODULE := nnstreamer_decoder_octet_stream
+LOCAL_SRC_FILES := $(NNSTREAMER_DECODER_PE_SRCS)
```
Comment (jaeyun-jung) [inline jni/Android.mk:223]: update nnstreamer.mk to define NNSTREAMER_DECODER_OS_SRCS

File: ext/nnstreamer/tensor_decoder/tensordec-octetstream.c (hunk 1 above)
Comment (gichan-jang) [reply ext/nnstreamer/tensor_decoder/tensordec-octetstream.c:98]: Thanks! updated.

File: jni/Android.mk (hunk 2 above)
Comment (gichan-jang) [reply jni/Android.mk]: Thanks. Updated.

File: ext/nnstreamer/tensor_decoder/tensordec-octetstream.c [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+#include <string.h>
+#include <glib.h>
+#include <nnstreamer_plugin_api_decoder.h>
+#include <nnstreamer_plugin_api.h>
+#include <nnstreamer_log.h>
+#include <nnstreamer_util.h>
+#include "tensordecutil.h"
+
+G_BEGIN_DECLS
```
Comment (dongju-chae) [inline ext/nnstreamer/tensor_decoder/tensordec-octetstream.c:25]: I'm not sure `G_BEGIN_DECLS` (which is `extern "c"`) is meaningful in a source file.

Comment (gichan-jang) [reply ext/nnstreamer/tensor_decoder/tensordec-octetstream.c]: Right. It's unnecessary. Removed. Thanks.

**Review by dongju-chae — APPROVED** [review]
LGTM

**Review by anyj0527 — APPROVED** [review]
LGTM ✔

**Review by jaeyun-jung — APPROVED** [review]

**Pull request merged**